### PR TITLE
Set up E2E tests

### PR DESCRIPTION
**Pull request by Flanker32**
Set up E2E tests to ensure maven plugin could work with the latest function host

### CODE REVIEW

File: appveyor_function_e2e.yml [hunk 1]
```diff
@@ -60,4 +60,5 @@ test_script:
 
 on_finish:
   - ps: | 
-      Stop-Process -Id $proc.Id -Erroraction Ignore
\ No newline at end of file
+      Stop-Process -Id $proc.Id -Erroraction Ignore
+
```
Comment (jdneo) [inline appveyor_function_e2e.yml]: remove space

File: appveyor_function_e2e.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+environment:
+  JAVA_HOME: "C:\\Program Files\\Java\\jdk1.8.0"
+  PYTHON: "C:\\Python35"
+  dotnetcore_version: 2.1.302
+  nodejs_version: "8"
+
+install:
+  - cmd: SET PATH=%PYTHON%;%PYTHON%\Scripts;%JAVA_HOME%\bin;%PATH%
+  - cmd: npm i -g azure-functions-core-tools@core --unsafe-perm true
+  - cmd: python -m pip install azure-cli
```
Comment (pragnagopa) [inline appveyor_function_e2e.yml]: Do not install core tools. Use the CLI from storage feed

File: appveyor_function_e2e.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      mvn archetype:generate -DarchetypeCatalog="local" -DarchetypeGroupId="com.microsoft.azure" -DarchetypeArtifactId="azure-functions-archetype" -DarchetypeVersion="$atchetypeVersion" -DgroupId="com.microsoft" -DartifactId="e2etestproject" -Dversion="1.0-SNAPSHOT" -Dpackage="com.microsoft" -DappRegion="westus" -DresourceGroup="e2etest-java-functions-group" -DappName="e2etest-java-functions" -B
+      mvn -f ".\e2etestproject\pom.xml" clean package
+      cd ..
+
+      # Run function host
+      $Env:AzureWebJobsScriptRoot = "$base\e2etestproject\e2etestproject\target\azure-functions\azure-functions-java-endtoendtests"
+      $Env:FUNCTIONS_WORKER_RUNTIME = "java"
+      $Env:AZURE_FUNCTIONS_ENVIRONMENT = "development"
+      $Env:Path = $Env:Path+";$base\Azure.Functions.Cli"
+      $proc = start-process -filepath "$base\Azure.Functions.Cli\func.exe" -WorkingDirectory "$base\e2etestproject\e2etestproject\target\azure-functions\e2etest-java-functions" -ArgumentList "host start" -PassThru
```
Comment (pragnagopa) [inline appveyor_function_e2e.yml]: Use environment variable AzureWebJobsScriptRoot for working directory

File: appveyor_function_e2e.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
+branches:
+  only:
+    - develop
+    
+environment:
+  JAVA_HOME: "C:\\Program Files\\Java\\jdk1.8.0"
+  PYTHON: "C:\\Python35"
```
Comment (pragnagopa) [inline appveyor_function_e2e.yml]: Is python needed?

**Comment by Flanker32** [comment]
@pragnagopa Comments resolved, besides, i've sent a invitations to you to join the group of appveyor, and then you could trigger this test [here](https://ci.appveyor.com/project/xscript/azure-maven-plugins-2e5yp) (For now it will build the e2etest branch, which will be changed to develop after merge)

**Pull request merged**